Sources/Controllers/2.0/OAFavoriteListViewController.mm

```diff
@@ -1157,21 +1168,27 @@ - (void)removeItemsFromUnsortedFavoriteItems:(NSArray *)selectedRows
             FavoriteTableGroup* groupData = [self.groupsAndFavorites objectAtIndex:selectedItem.section];
             OAFavoriteItem* item = [groupData.groupItems objectAtIndex:dataIndex];
             
-            [self.favoriteTableView beginUpdates];
             app.favoritesCollection->removeFavoriteLocation(item.favorite);
             [groupData.groupItems removeObjectAtIndex:dataIndex];
-            [self.favoriteTableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:[NSIndexPath indexPathForRow:selectedItem.row inSection:selectedItem.section]] withRowAnimation:UITableViewRowAnimationLeft];
-            [self.favoriteTableView endUpdates];
+            
+            if (groupData.isOpen)
+            {
+                [self.favoriteTableView beginUpdates];
+            
+                [self.favoriteTableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:[NSIndexPath indexPathForRow:selectedItem.row inSection:selectedItem.section]] withRowAnimation:UITableViewRowAnimationLeft];
+                [self.favoriteTableView endUpdates];
+            }
             [app saveFavoritesToPermamentStorage];
         }
     }
+    [_selectedUnsortedItems removeAllObjects];
     [self editButtonClicked:nil];
 }
 
-- (void)removeFavoriteItems:(NSArray *)selectedRows
+- (void) removeFavoriteItems:(NSArray *)selectedRows
```

Comment:
Looks like half of work done. I think we need handle selection via array for both modes sorted / unsorted: _selectedItems